polling.go

```diff
@@ -29,27 +29,34 @@ func newPollingProcessor(config Config, requestor *requestor) *pollingProcessor
 func (pp *pollingProcessor) Start(closeWhenReady chan<- struct{}) {
 	pp.config.Logger.Printf("Starting LaunchDarkly polling processor with interval: %+v", pp.config.PollInterval)
 	go func() {
+		var readyOnce sync.Once
+		notifyReady := func() {
+			readyOnce.Do(func() {
+				close(closeWhenReady)
+			})
+		}
+		// Ensure we stop waiting for initialization if we exit, even if initialization fails
+		defer notifyReady()
+
 		for {
 			select {
 			case <-pp.quit:
 				pp.config.Logger.Printf("Polling Processor closed.")
-				close(closeWhenReady)
 				return
 			default:
 				then := time.Now()
 				err := pp.poll()
 				if err == nil {
 					pp.setInitializedOnce.Do(func() {
 						pp.isInitialized = true
-						close(closeWhenReady)
+						notifyReady()
 					})
 				} else {
 					pp.config.Logger.Printf("ERROR: Error when requesting feature updates: %+v", err)
 					if hse, ok := err.(*HttpStatusError); ok {
 						if hse.Code == 401 {
 							pp.config.Logger.Printf("ERROR: Received 401 error, no further polling requests will be made since SDK key is invalid")
-							close(closeWhenReady)
-							return
+							notifyReady()
```

Comment:
You still should return here, no?
If you got 401 you don't have anything to do, just break the looping. WDY?